aiohttp/streams.py

```diff
@@ -220,6 +220,9 @@ def feed_eof(self) -> None:
             self._eof_waiter = None
             set_result(waiter, None)
 
+        if self._protocol._reading_paused:
```

Comment:
Leaving an open conversation so this doesn't merge before https://github.com/aio-libs/aiohttp/pull/10171#issuecomment-2548475830